### ISSUE TITLE
Add support for registering data from various Dataverse installations

### ISSUE THREAD

**Issue by amoeba** [whole-tale/dashboard#269]
A few of us talked with @pdurbin at the 2018 workshop about adding support for Dataverse's API in WholeTale's Data Registration feature. This looks really easy. One note is that there are many Dataverse installations and no central location to query across all of them at once but they appear to all follow a common convention so I think we can easily support Harvard's installation but with only a bit more effort, any installation:

- WholeTale sees the DOI: 10.7910/DVN/TJCLKP
- And follows it to: https://dataverse.harvard.edu/dataset.xhtml?persistentId=doi:10.7910/DVN/TJCLKP
- Then we can try the API: https://dataverse.harvard.edu/api/datasets/:persistentId?persistentId=doi:10.7910/DVN/TJCLKP

which gives a nicely-structured response body:

```json
{
  "status": "OK",
  "data": {
    "id": 3035124,
    "identifier": "DVN/TJCLKP",
    "persistentUrl": "https://doi.org/10.7910/DVN/TJCLKP",
...snip...
   "files": [
        {
          "label": "2017-07-31.tab",
...snip...
```

which has the necessary things: Dataset title, and for all files: filename, and size.

**Comment by pdurbin** [comment]
@amoeba yes! Great workshop! The detail I mentioned is that to actually download the file you will need the database id of the file. For example, for the following file the database id is "3040230":

```
"files": [
  {
    "label": "2017-07-31.tab",
    "restricted": false,
    "version": 2,
    "datasetVersionId": 127386,
    "dataFile": {
      "id": 3040230,
```

One you have the database id you can download the file, assuming it doesn't have any restrictions, at URLs like this:

- https://dataverse.harvard.edu/api/access/datafile/3040230
- https://dataverse.harvard.edu/api/access/datafile/3040230?format=original

You can learn more about the available query parameters, such as "format" above, at http://guides.dataverse.org/en/4.9.2/api/dataaccess.html

And yes, while we're using Harvard Dataverse as an example, It would be great to support any installation of Dataverse. You can get a list of Dataverse installations in JSON format at https://services.dataverse.harvard.edu/miniverse/map/installations-json

**Comment by amoeba** [comment]
Perfect, that's the missing piece.

> You can get a list of Dataverse installations in JSON format at services.dataverse.harvard.edu/miniverse/map/installations-json

Nice! thanks

**Comment by pdurbin** [comment]
@amoeba you're quite welcome. When you get a chance, please do go ahead and create an issue over at https://github.com/IQSS/dataverse/issues called something like "add 'Explore in Whole Tale' button" for the other integration we talked about. I can leave a detailed comment in the issue you open but it amounts to adding a blurb in the External Tools section of the Dataverse Installation Guide with a link to instructions hosted by Whole Tale on how to add the external tool. This is a bit abstract so please see the "Data Explorer" example below:

![screen shot 2018-09-17 at 11 08 32 am](https://user-images.githubusercontent.com/21006/45647720-9014ab80-ba94-11e8-8786-b395e4fabb08.png)

The screenshot above is from http://guides.dataverse.org/en/4.9.2/installation/external-tools.html

Thanks!

**Comment by amoeba** [comment]
Right on, thanks @pdurbin 	! Issue filed.

**Comment by pdurbin** [comment]
@amoeba thanks for opening https://github.com/IQSS/dataverse/issues/5097 about adding an "external tool" to Dataverse for Whole Tale

With regard to registering data from Dataverse, this hopefully just got easier because https://github.com/IQSS/dataverse/pull/5169 was merged yesterday and allows you to use Schema.org JSON-LD metadata embedded in the "meta" tag of the dataset landing page (which you are redirected to from the DOI) to get a list of public files that you can download from the dataset. Here are some screenshots:

![screen shot 2018-11-08 at 11 46 53 am](https://user-images.githubusercontent.com/21006/48213882-c3650f80-e34c-11e8-932d-6548959cb5b1.png)

![screen shot 2018-11-08 at 11 46 58 am](https://user-images.githubusercontent.com/21006/48213881-c3650f80-e34c-11e8-9f54-01516410919d.png)

You can use a tool like https://github.com/scrapinghub/extruct to get the Schema.org JSON-LD from the "meta" tag.

The following curl command won't work on Harvard Dataverse until we upgrade to whatever version of Dataverse comes after 4.9.4 but here's a preview: `curl 'https://dataverse.harvard.edu/api/datasets/export?exporter=schema.org&persistentId=doi:10.7910/DVN/TJCLKP' | jq '.distribution[].contentUrl'`

If you'd like to play around with this you are welcome to use http://phoenix.dataverse.org which has pre-release code deployed. The data coming out from "distribution" will look something like this:

```
{
  "@type": "DataDownload",
  "name": "trees.png",
  "fileFormat": "image/png",
  "contentSize": 8361,
  "description": "",
  "@id": "https://doi.org/10.5072/FK2/XL9AXZ/0HDKSN",
  "identifier": "https://doi.org/10.5072/FK2/XL9AXZ/0HDKSN",
  "contentUrl": "http://phoenix.dataverse.org/api/access/datafile/100"
}
```

@craig-willis I see you opened https://github.com/whole-tale/girder_wholetale/issues/172 which is awesome.

It was nice talking to both of you at the open science infra meeting the other day! This one: https://github.com/whole-tale/whole-tale/issues/43

If you have any questions about any of this, please let me know!
